lib/spack/spack/util/compression.py

```diff
@@ -235,7 +247,15 @@ def _7zip(archive_file):
     Args:
         archive_file (str): absolute path of file to be unarchived
     """
+    # 7zip will decompress and extract .txz atomically
+    # but not .tar.xz, rename tar.xz
+    # to .txz for atomic decompression
+    if 'tar.xz' in archive_file:
+        file_path = archive_file.strip('.tar.xz')
+        os.rename(archive_file, file_path + '.txz')
```

Comment:
`archive_file` may refer to a reused file in the download cache, so I think we'd want to actually copy and rename this (otherwise we may rename the file inside of the cache).